### ISSUE TITLE
getChunkInfo doesn't assumes all the files in that chunk has the same location.

### ISSUE THREAD

**Issue by ticoann** [dmwm/WMCore#6845]
In following case, files are not in the same location T0_CH_CERN_Disk and T1_ES_PIC_Disk

https://cmsweb.cern.ch/acdcserver/_design/ACDC/_view/byCollectionName?key=%22jen_a_SMP-RunIISummer15wmLHEGS-00007_00004_v0__160419_232135_4008%22&reduce=false&include_docs=true

**Comment by ticoann** [comment]
Following code need to be checked  and location need be be grouped

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ACDC/DataCollectionService.py#L213

Instead use some thing like this

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ACDC/DataCollectionService.py#L139

**Comment by ticoann** [comment]
@amaltaro, Alan I am not completely sure whether this is the bug. Could you check what I said makes sense?

**Comment by amaltaro** [comment]
Hmm, I think it makes sense. I know there are several flaws in the ACDC code, but this one is unexpected... Each resubmission workflows creates a *single* workqueue element, so the location could be a *union* of all locations of all files (but then, I still think there would be jobs failing).

Otherwise, we make it return chunks but then we need to rework the ResubmitBlock policy (possible creating more workqueue elements based by location).

sigh, these things are on my todo list for quite a long time (including a PR opened since 2015 :()

**Comment by ticoann** [comment]
> Hmm, I think it makes sense. I know there are several flaws in the ACDC code, but this one is unexpected... Each resubmission workflows creates a single workqueue element, so the location could be a union of all locations of all files (but then, I still think there would be jobs failing).

I thought it is more of intersection since elements need to be matched with site resources. Since this is ACDC the file should only resides only one site unless it fails on first task. So we could just group the files with the same site to create the block

**Comment by amaltaro** [comment]
and what if we have 3 files in distinct locations? there is no intersection and there would be no block location in that case. Yes, an acdc file should reside in a single location, no matter what.

That's what I don't think will work, from what I know there is a single "block" for ACDC requests (which corresponds to a single WQE), no?

**Comment by ticoann** [comment]
I mean if intersection is not empty we group them with one block if there is no intersection we create new block for new file.  Basically chunkFileset is doing that.

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ACDC/DataCollectionService.py#L138

I wasn't sure whether this is a bug since I wasn't sure what is the condition for 

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/Policy/Start/ResubmitBlock.py#L95

I was wondering whether this implies all the files in the same site.

**Comment by amaltaro** [comment]
> I mean if intersection is not empty we group them with one block if there is no intersection we create new block for new file. 

Yes, I agree! It should be calling chunkFileset() instead of getChunkInfo()

Sorry, only now I see it can create more than one block:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/Policy/Start/ResubmitBlock.py#L60

I think the current code would create a single block/WQE with the location pointing to the sorted locations output, regardless of the real location of all the files.

The question is, do you see it causing issues in production or you just stumbled upon this code?

**Comment by amaltaro** [comment]
I'd like to work on some ACDC issues, so I'll add this one to my list as well.

**Comment by amaltaro** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [getChunkInfo doesn't assumes all the files in that chunk has the same location](https://trello.com/c/JWyQSKrd/13-getchunkinfo-doesn-t-assumes-all-the-files-in-that-chunk-has-the-same-location)
